src/reader/buffered_reader.rs

```diff
@@ -12,6 +12,216 @@ use crate::events::Event;
 use crate::name::QName;
 use crate::reader::{is_whitespace, BangType, ReadElementState, Reader, XmlSource};
 
+macro_rules! impl_buffered_source {
```

Comment:
I don't know... Having macro in this place first of all indicates that it implements a buffering reader. Because "buffering" trait of this macro seems to me more important, than "async" trait, it is defined here. Moving it into another module will break the explicit link with that usage place, as I think

Reply:
It's your choice in the end. Keeping it here would work too.